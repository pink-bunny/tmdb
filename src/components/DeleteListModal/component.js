import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Typography } from 'antd';

const DeleteListModal = ({
  name,
  handleTriggerClick,
  modalVisible,
  hideModal,
  deleteList,
  error,
  triggerComponent: TriggerComponent,
  triggerProps
}) => (
  <>
    <TriggerComponent
      {...triggerProps} /* eslint-disable-line react/jsx-props-no-spreading */
      onClick={handleTriggerClick}
    />

    <Modal
      visible={modalVisible}
      onCancel={hideModal}
      onOk={deleteList}
      title="Delete list"
    >
      {error && (
        <Typography.Paragraph
          type="danger"
          style={{ marginBottom: '16px' }}
        >
          {error}
        </Typography.Paragraph>
      )}

      {`Do you want to delete ${name} list?`}
    </Modal>
  </>
);

DeleteListModal.propTypes = {
  name: PropTypes.string.isRequired,
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  deleteList: PropTypes.func.isRequired,
  error: PropTypes.string,
  handleTriggerClick: PropTypes.func.isRequired,
  triggerProps: PropTypes.shape({
    onClick: PropTypes.func
  }),
  triggerComponent: PropTypes.func.isRequired
};

DeleteListModal.defaultProps = {
  triggerProps: null,
  error: ''
};

export default DeleteListModal;
