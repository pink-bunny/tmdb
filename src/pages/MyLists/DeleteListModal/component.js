import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

const DeleteListModal = ({ deleteListModal }) => (
  <Icon type="delete" onClick={deleteListModal} />
);

DeleteListModal.propTypes = {
  deleteListModal: PropTypes.func.isRequired
};

export default DeleteListModal;
