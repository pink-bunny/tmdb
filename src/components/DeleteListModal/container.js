import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import withModal from '../../hoc/withModal';
import { myListsDeleteListError } from '../../../state/my_lists/selectors';
import {
  deleteMyList as deleteMyListAction
} from '../../../state/my_list/actions';
import DeleteListModal from './component';

class DeleteListModalContainer extends React.Component {
  deleteList = () => {
    const { deleteMyList, id, hideModal } = this.props;

    deleteMyList(id, hideModal);
  };

  render() {
    const {
      id,
      name,
      modalVisible,
      showModal,
      hideModal,
      error,
      triggerComponent,
      triggerProps
    } = this.props;

    return (
      <DeleteListModal
        id={id}
        name={name}
        error={error}
        deleteList={this.deleteList}
        modalVisible={modalVisible}
        handleTriggerClick={showModal}
        hideModal={hideModal}
        triggerComponent={triggerComponent}
        triggerProps={triggerProps}
      />
    );
  }
}

DeleteListModalContainer.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  deleteMyList: PropTypes.func.isRequired,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  name: PropTypes.string.isRequired,
  error: PropTypes.string,
  triggerProps: PropTypes.shape({
    onClick: PropTypes.func
  }),
  triggerComponent: PropTypes.func.isRequired
};

DeleteListModalContainer.defaultProps = {
  error: '',
  triggerProps: null
};

const mapStateToProps = (state) => ({
  error: myListsDeleteListError(state)
});

const mapDispatchToPtops = {
  deleteMyList: deleteMyListAction
};

export default connect(mapStateToProps, mapDispatchToPtops)(withModal(DeleteListModalContainer));
