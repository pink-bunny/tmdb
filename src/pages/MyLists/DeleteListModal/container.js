import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import withModal from '../../../hoc/withModal';
import { deleteMyListError } from '../../../../state/my_lists/selectors';
import {
  deleteMyList as deleteMyListAction
} from '../../../../state/my_lists/actions';
import DeleteListModal from './component';

class DeleteListModalContainer extends React.Component {
  deleteList = () => {
    const { deleteMyList, id, hideModal } = this.props;

    return (
      deleteMyList(id, hideModal)
    );
  };

  render() {
    const {
      id,
      name,
      modalVisible,
      showModal,
      hideModal,
      error
    } = this.props;

    return (
      <DeleteListModal
        id={id}
        name={name}
        error={error}
        deleteList={this.deleteList}
        modalVisible={modalVisible}
        showModal={showModal}
        hideModal={hideModal}
      />
    );
  }
}

DeleteListModalContainer.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  deleteMyList: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  error: PropTypes.string
};

DeleteListModalContainer.defaultProps = {
  error: ''
};

const mapStateToProps = (state) => ({
  error: deleteMyListError(state)
});

const mapDispatchToPtops = {
  deleteMyList: deleteMyListAction
};

export default connect(mapStateToProps, mapDispatchToPtops)(withModal(DeleteListModalContainer));
