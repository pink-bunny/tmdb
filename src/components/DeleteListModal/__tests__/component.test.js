import React from 'react';
import { shallow } from 'enzyme';

import DeleteListModal from '../component';

describe('DeleteListModal component', () => {
  const requiredProps = {
    name: 'Test list',
    modalVisible: false,
    hideModal: jest.fn(),
    handleTriggerClick: jest.fn(),
    deleteList: jest.fn(),
    triggerComponent: jest.fn(() => <div>Mock button</div>)
  };

  it('matches snapshot with required props', () => {
    const wrapper = shallow(<DeleteListModal {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with error', () => {
    const props = {
      ...requiredProps,
      error: 'Delete error'
    };
    const wrapper = shallow(<DeleteListModal {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
