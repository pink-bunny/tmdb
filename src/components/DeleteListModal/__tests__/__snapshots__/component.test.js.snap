// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteListModal component matches snapshot with optional props 1`] = `
<Fragment>
  <mockConstructor
    onClick={[MockFunction]}
    type="button"
  >
    Mock button
  </mockConstructor>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Delete list"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Paragraph
      style={
        Object {
          "marginBottom": "16px",
        }
      }
      type="danger"
    >
      Delete error
    </Paragraph>
    Do you want to delete Test list list?
  </Modal>
</Fragment>
`;

exports[`DeleteListModal component matches snapshot with required props 1`] = `
<Fragment>
  <mockConstructor
    onClick={[MockFunction]}
  />
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Delete list"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    Do you want to delete Test list list?
  </Modal>
</Fragment>
`;
